File: 2021/day1/valid-anagram/Valid_anagram.cpp
class Solution //By CyFun
{
public:
    bool isAnagram(string s, string t)
    {
        vector<int> temp(26, 0);
        if (s.length() != t.length())
            return false;
        for (int i = 0; i < s.length(); i++)
        {
            temp[(int)s[i] - (int)'a']++;
            temp[(int)t[i] - (int)'a']--;
        }
        for (int i = 0; i < 26; i++)
        {
            if (temp[i] != 0)
                return false;
        }
        return true;
    }
};

File: 2021/Day9/Valid Anagram/valid_anagram.py
class Solution:
    	def isAnagram(self, s: str, t: str) -> bool:
		ns, nt = len(s), len(t)
		if ns != nt: return False
		if ns == 0 and nt == 0: return True
		elif ns == 0 or nt == 0: return False
		return self.isAnagram(s.replace(s[0], ''), t.replace(s[0], ''))